frontend/src/container/MetricsApplication/MetricsPageQueries/OverviewQueries.ts

```diff
@@ -22,79 +24,62 @@ import {
 export const latency = ({
 	servicename,
 	tagFilterItems,
+	spanMetricFlg = false,
 }: LatencyProps): QueryBuilderData => {
-	const autocompleteData: BaseAutocompleteData[] = [
-		{
-			key: WidgetKeys.DurationNano,
-			dataType: DataType.FLOAT64,
-			isColumn: true,
-			type: MetricsType.Tag,
-		},
-		{
-			key: WidgetKeys.DurationNano,
-			dataType: DataType.FLOAT64,
-			isColumn: true,
-			type: MetricsType.Tag,
-		},
+	const newAutoCompleteData: BaseAutocompleteData = {
+		key: spanMetricFlg
+			? WidgetKeys.Signoz_latency_bucket
+			: WidgetKeys.DurationNano,
+		dataType: DataType.FLOAT64,
+		isColumn: true,
+		type: spanMetricFlg ? null : MetricsType.Tag,
+	};
+
+	const autocompleteData: BaseAutocompleteData[] = Array(3).fill(
+		newAutoCompleteData,
+	);
+
+	const filterItem: TagFilterItem[] = [
 		{
-			key: WidgetKeys.DurationNano,
-			dataType: DataType.FLOAT64,
-			isColumn: true,
-			type: MetricsType.Tag,
+			id: '',
+			key: {
+				key: spanMetricFlg ? WidgetKeys.Service_name : WidgetKeys.ServiceName,
+				dataType: DataType.STRING,
+				type: spanMetricFlg ? MetricsType.Resource : MetricsType.Tag,
+				isColumn: !spanMetricFlg,
+			},
+			op: spanMetricFlg ? OPERATORS.IN : OPERATORS['='],
+			value: spanMetricFlg ? [servicename] : servicename,
 		},
+		...tagFilterItems,
 	];
 
-	const filterItems: TagFilterItem[][] = [
-		[
-			{
-				id: '',
-				key: {
-					key: WidgetKeys.ServiceName,
-					dataType: DataType.STRING,
-					type: MetricsType.Tag,
-					isColumn: true,
-				},
-				op: OPERATORS['='],
-				value: `${servicename}`,
-			},
-			...tagFilterItems,
-		],
-		[
-			{
-				id: '',
-				key: {
-					key: WidgetKeys.ServiceName,
-					dataType: DataType.STRING,
-					type: MetricsType.Tag,
-					isColumn: true,
-				},
-				op: OPERATORS['='],
-				value: `${servicename}`,
-			},
-			...tagFilterItems,
-		],
-		[
-			{
-				id: '',
-				key: {
-					key: WidgetKeys.ServiceName,
-					dataType: DataType.STRING,
-					type: MetricsType.Tag,
-					isColumn: true,
-				},
-				op: OPERATORS['='],
-				value: `${servicename}`,
+	if (spanMetricFlg) {
+		filterItem.push({
```

Comment:
This should be done for traces as well. The difference is `key` part. It will be
```
key: {
	dataType: DataType.STRING,
	isColumn: true,
	key: 'name',
	type: MetricsType.Tag,
}
```

Reply:
Okay got it.